client/me/purchases/pre-cancellation-dialog/index.tsx

```diff
@@ -0,0 +1,226 @@
+
+	/**
+	 * Click events, buttons tracking and action.
+	 */
+	const clickCancelPlan = () => {
+		removePlan();
+	};
+
+	const clickCloseDialog = () => {
+		closeDialog();
+	};
+
+	/**
+	 * Domain redirect copy
+	 */
+	const domainFeature =
+		hasDomain && primaryDomain && wpcomURL ? (
+			<p>Warning: domain traffic will be redirected</p>
+		) : null;
+
+	/**
+	 * Dialog buttons
+	 */
+	const buttons = [
+		{
+			action: 'cancel',
+			label: translate( 'Cancel my plan' ),
+			onClick: clickCancelPlan,
+		},
+		{
+			action: 'close',
+			label: translate( 'Keep my plan' ),
+			onClick: clickCloseDialog,
+			isPrimary: true,
+		},
+	];
+
+	/**
+	 * Render dialog bottom text:
+	 * - Link to contact support.
+	 * - If refundable, the amount to be refunded.
+	 */
+	const RenderFooterText = () => {
+		const { refundText } = purchase;
+
+		return (
+			<div className="pre-cancellation-dialog--footer">
+				<div className="pre-cancellation-dialog--footer--support">
+					<strong className="pre-cancellation-dialog--footer--support-information">
+						{ ! isRefundable( purchase ) && maybeWithinRefundPeriod( purchase )
+							? translate(
+									'Have a question? Want to request a refund? {{contactLink}}Ask a Happiness Engineer!{{/contactLink}}',
+									{
+										components: {
+											contactLink: <a href={ CALYPSO_CONTACT } />,
+										},
+									}
+							  )
+							: translate(
+									'Have a question? {{contactLink}}Ask a Happiness Engineer!{{/contactLink}}',
+									{
+										components: {
+											contactLink: <a href={ CALYPSO_CONTACT } />,
+										},
+									}
+							  ) }
+					</strong>
+				</div>
+				<div className="pre-cancellation-dialog--footer--refund">
+					{ hasAmountAvailableToRefund( purchase ) &&
+						translate( '%(refundText)s to be refunded', {
+							args: { refundText },
+							context: 'refundText is of the form "[currency-symbol][amount]" i.e. "$20"',
+						} ) }
+				</div>
+			</div>
+		);
+	};
+
+	/**
+	 * Return the list of features that the user will lose by canceling their plan.
+	 */
+	const FeaturesList = () => {
+		if ( typeof productSlug !== 'string' ) {
+			return null;
+		}
+
+		const planFeatures = getPlanCancellationFeatures( productSlug );
+
+		return (
+			<>
+				<p>{ subTitle }</p>
+				<ul
+					className={
+						'pre-cancellation-dialog__list-plan-features' +
+						( domainFeature ? ' --with-domain-feature' : '' )
+					}
+				>
+					{ domainFeature && (
+						<li key="redirect-domain">
+							<Gridicon
+								className="pre-cancellation-dialog__item-cross-small"
+								size={ 24 }
+								icon="cross-small"
+							/>
+							{ domainFeature }
+						</li>
+					) }
+					<li key="debug-refundable">
+						<Gridicon
+							className="pre-cancellation-dialog__item-cross-small"
+							size={ 24 }
+							icon="cross-small"
+						/>
+						{ isPurchaseRefundable ? 'Refundable: yes' : 'Refundable: no' }
```

Comment:
Does this text need to be localized?

Reply:
No. The same reasoning below also applies here. (Now you can see sometimes I actually read a PR backward ...)